llvm/lib/Analysis/DXILResource.cpp

```diff
@@ -770,6 +770,50 @@ void DXILBindingMap::print(raw_ostream &OS, DXILResourceTypeMap &DRTM,
   }
 }
 
+SmallVector<dxil::ResourceBindingInfo>
+DXILBindingMap::findByUse(const Value *Key) const {
+  const PHINode *Phi = dyn_cast<PHINode>(Key);
+  if (Phi) {
```

Comment:
Best to limit the scope by declaring `Phi` in the `if()` here
```suggestion
  if (const PHINode *Phi = dyn_cast<PHINode>(Key)) {
```